packages/schema/parse/src/version_compare/setUtils.ts

```diff
@@ -0,0 +1,32 @@
+export enum SetComparisionType {
+  EQUAL,
+  SUBSET,
+  DIFFERENT,
+}
+
+export function union<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1, ...set2]);
+}
+
+export function intersection<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1].filter((x) => set2.has(x)));
+}
+
+export function difference<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return new Set([...set1].filter((x) => !set2.has(x)));
+}
+
+export function symmentric_difference<T>(set1: Set<T>, set2: Set<T>): Set<T> {
+  return difference(union(set1, set2), intersection(set1, set2));
+}
+
+export function compareSets<T>(set1: Set<T>, set2: Set<T>): SetComparisionType {
+  const sd: Set<T> = symmentric_difference(set1, set2);
```

Comment:
A faster way would be:
```typescript
if(set1.length > set2.length) {
  //No need to check anything else in this case, so we wan't to return early
  return SetComparisionType.DIFFERENT;
}

for(const item of set1) {
  if(!set2.has(item)) {
    return SetComparisionType.DIFFERENT;
  }
}

if(set1.length === set2.length) {
  return SetComparisionType.EQUAL;
} else {
  return SetComparisionType.SUBSET 
}
```

Reply:
No need for other functions

Reply:
Makes sense, I was following mathematical theorems but this looks better computationally :)